### ISSUE TITLE
Error type checking TypedDict (when total=False)

### ISSUE THREAD

**Issue by mariusmuja** [microsoft/pyright#1798]
**Describe the bug**

The following code fails to typecheck:

```python
from typing import TypedDict


Config = TypedDict("Config", {
        "foo": int,
        "bar": str,
    }, total = False
)

def test(c: Config):
    pass

def main(c: Config):
    if "foo" in c:
        test(c)
```

with the error:
```
example.py:18:14 - error: Argument of type "Config" cannot be assigned to parameter "c" of type "Config" in function "test"
    "foo" is not required in "Config" (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Either.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pyright and pylance.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.136, which I just published. It will also be included in the next release of pylance.
